### ISSUE TITLE
A look at space group functionality in diffpy.structure

### ISSUE THREAD

**Issue by hakonanes** [pyxem/orix#69]
@dnjohnstone suggested in a review comment https://github.com/pyxem/orix/pull/66#discussion_r440026545 that we could start looking at the space group/symmetry stuff in diffpy.structure, e.g. the `SpaceGroup` class: https://github.com/diffpy/diffpy.structure/blob/master/src/diffpy/structure/spacegroupmod.py#L194. After briefly looking at the `SpaceGroup` class now when working on #67 to get symmetrically equivalent atom positions in a unit cell (via `diffpy.structure.symmetryutilities.expandPosition`), this is what an object itself contains:

```python
>>> from diffpy.structure.spacegroups import GetSpaceGroup
>>> sg225 = GetSpaceGroup(225)
>>> print(sg225.__class__)
diffpy.structure.spacegroupmod.SpaceGroup
>>> print(sg225.__dict__)
{'number': 225,
 'num_sym_equiv': 192,
 'num_primitive_sym_equiv': 48,
 'short_name': 'Fm-3m',
 'point_group_name': 'PGm3barm',
 'crystal_system': 'CUBIC',
 'pdb_name': 'F 4/m -3 2/m',
 'symop_list': [<diffpy.structure.spacegroupmod.SymOp at 0x7f01817c7dd0>,
  <diffpy.structure.spacegroupmod.SymOp at 0x7f01817c7e10>,
  <diffpy.structure.spacegroupmod.SymOp at 0x7f01817c7e50>,
  <diffpy.structure.spacegroupmod.SymOp at 0x7f01817c7e90>,
  [...]
  <diffpy.structure.spacegroupmod.SymOp at 0x7f01817d1e50>]}
```

Definition of rotation matrices and rotation vectors for the space groups: https://github.com/diffpy/diffpy.structure/blob/master/src/diffpy/structure/spacegroupmod.py

Definition of the space groups: https://github.com/diffpy/diffpy.structure/blob/master/src/diffpy/structure/mmlibspacegroups.py

How does the `orix.quaternion.symmetry.Symmetry` class compare to the `diffpy.structure.spacegroupmod.SymOp` class? It seems like `Symmetry` has more functionality, @pc494? I haven't used the `Symmetry` class yet, so therefore I cannot say what we gain by starting using the `SymOp` and `SpaceGroup` classes.

**Comment by dnjohnstone** [comment]
I think Symmetry does contain more than SymOp right now and I don't think we would gain much from changing that. My reason for bringing this up is that, given how small the pythonic crystallography community is, it seems sensible to me that we try to help that community to converge around one library for the foundational stuff like symmetry operations.

Given how frequently that stuff is needed the potential user base could be quite big and it would be good to have as many people using that core stuff as possible to minimise the chances of bugs being missed.

There are related things e.g. https://github.com/diffpy/pyobjcryst and https://github.com/cctbx/cctbx_project and https://github.com/materialsproject/pymatgen

Originally we used pymatgen in pyxem packages but found it was excessively large for what we actually wanted, similarly with cctbx there's lots there that we don't want/need.

So diffpy.structure seemed like the best lightweight and compartmentalized option to build on.

In other words, I think we should be thinking about what functionality that we've currently put in orix/diffsims that might be better located in diffpy.structure - then we could reach out to the diffpy developers to see if we might contribute to add momentum to that package.

Does that make sense?

**Comment by pc494** [comment]
My experience so far with `diffpy` is that there is some functionality that I wanted that was not yet there, especially in terms of building bigger unit cells. For this issue, I think `orix` is the way to go, in general, I think if we have ideas that make sense to go into `diffpy` we should see reach out to them directly.

**Comment by dnjohnstone** [comment]
@hakonanes - do we think this still needs to be open?

**Comment by hakonanes** [comment]
> @hakonanes - do we think this still needs to be open?

Ah, sorry, no. We can continue the discussion in #87.
